librz/analysis/p/analysis_x86_cs.c

```diff
@@ -3118,45 +3123,53 @@ static int cs_len_prefix_opcode(uint8_t *item) {
 }
 
 static int analop(RzAnalysis *a, RzAnalysisOp *op, ut64 addr, const ut8 *buf, int len, RzAnalysisOpMask mask) {
-	static int omode = 0;
-#if USE_ITERZ_API
-	static
-#endif
-		cs_insn *insn = NULL;
-	int mode = (a->bits == 64) ? CS_MODE_64 : (a->bits == 32) ? CS_MODE_32
-		: (a->bits == 16)                                 ? CS_MODE_16
-								  : 0;
+	X86CSContext *ctx = (X86CSContext *)a->plugin_data;
+
+	int mode = 0;
```

Comment:
Extract this in a small inline function, I think.

Reply:
You mean to extract the context ?

Reply:
No, mode handling.